lib/src/chatbox.dart

```diff
@@ -52,6 +53,13 @@ class MessageActionEvent {
     message = Message.fromJson(json['message']);
 }
 
+enum UrlNavigationAction { allow, deny }
+
+class UrlNavigationRequest {
+  final url;
```

Comment:
I Updated the UrlNavigationRequest it now had 3 properties:
```
final Uri? url;
final String rawUrl;
final bool isValidUrl;
```

This is to make sure that if the url is not a valid uri that it still is possible to pass the UrlNavigationRequest without throwing exceptions. 

I'm also thinking that it might be better to just pass the url as a String, this way any error handling or conversion to Uri Objects can be handled by the specific app using the plugin. 
What do you feel is best?

Reply:
OK, so in this case let's use String for the url, and the validation is done by the user's code.
I thought that the Uri type could be the more appropriate one, but given that there could be some cases where the url is not a valid Uri, then my previous assumption was wrong.